public/components/integrations/components/setup_integration.tsx

```diff
@@ -635,29 +655,81 @@ export function SetupIntegrationPage({ integration }: { integration: string }) {
   const updateConfig = (updates: Partial<IntegrationSetupInputs>) =>
     setConfig(Object.assign({}, integConfig, updates));
 
-  return (
-    <EuiPage>
-      <EuiPageBody>
+  if (renderType === 'page') {
+    return (
+      <>
         <EuiPageContent>
           <EuiPageContentBody>
             {showLoading ? (
               <LoadingPage />
             ) : (
-              <SetupIntegrationForm
+              <SetupIntegrationFormInputs
                 config={integConfig}
                 updateConfig={updateConfig}
                 integration={template}
                 setupCallout={setupCallout}
+                lockConnectionType={forceConnectionType !== undefined}
               />
             )}
           </EuiPageContentBody>
         </EuiPageContent>
-        <SetupBottomBar
-          config={integConfig}
-          integration={template}
-          loading={showLoading}
-          setLoading={setShowLoading}
-          setSetupCallout={setSetupCallout}
+        <EuiBottomBar>
+          <SetupBottomBar
+            config={integConfig}
+            integration={template}
+            loading={showLoading}
+            setLoading={setShowLoading}
+            setSetupCallout={setSetupCallout}
+            unsetIntegration={unsetIntegration}
+          />
+        </EuiBottomBar>
+      </>
+    );
+  } else if (renderType === 'flyout') {
```

Comment:
Bottom bar doesn't render in a flyout if we keep the existing EuiPage component, so I added multiple render types that swap the top-level components between those for an independent page or those for a flyout body. Slightly annoying but not sure what a better way is.